main/src/ca/uwaterloo/flix/language/phase/Desugar.scala

```diff
@@ -555,6 +555,14 @@ object Desugar {
       val e2 = visitExp(exp2)
       Expr.LetRec(ident, ann, mod, e1, e2, loc)
 
+    case WeededAst.Expr.LocalDef(ann, ident, fparams, dtpe, deff, exp1, exp2, loc) =>
```

Comment:
I don't think we should have dtpe and deff after this actually. shouldn't it just become `def (...) = (body: dtpe \ deff)` like lets do in this phase?

Reply:
I'm not quite sure I understand what you mean :)

Reply:
I think he is questioning whether we should allow/support return type/return effect declarations.

Reply:
But I don't see why we should not support them.

Reply:
no i mean that `let x: t = e` becomes `let x = (e: t)` in this phase

I don't see why we shouldn't do the same translation here. I was looking at why `let` didn't have `dtpe` and `deff` and that's the reason. So to be consistent we should do the same or change let

Reply:
Ah I see, that way we also avoid generating the type constraint `t = e` manually

Reply:
yeah 👍